django/pyrog/models.py

```diff
@@ -0,0 +1,180 @@
+    definition_id = models.TextField()
+    logical_reference = models.TextField()
+    primary_key_owner = models.ForeignKey(
+        "Owner", blank=True, null=True, on_delete=models.CASCADE
+    )  # TODO: Why nullable ?
+
+    updated_at = models.DateTimeField(auto_now=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+
+class Credential(models.Model):
+    class Dialect(models.TextChoices):
+        MSSQL = "MSSQL"
+        POSTGRESQL = "POSTGRES"
+        ORACLE = "ORACLE"
+        SQLLITE = "SQLLITE"
+
+    id_ = models.TextField(name="id", primary_key=True, default=cuid, editable=False)
+    source = models.OneToOneField(Source, on_delete=models.CASCADE)
+    host = models.TextField()
+    port = models.IntegerField()
+    database = models.TextField()
+    login = models.TextField()
+    password = models.TextField()
+    model = models.TextField(choices=Dialect.choices)
+
+    updated_at = models.DateTimeField(auto_now=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+
+class Attribute(models.Model):
+    id_ = models.TextField(name="id", primary_key=True, default=cuid, editable=False)
+    path = models.TextField()
+    slice_name = models.TextField(blank=True, default="")
+    definition_id = models.TextField()
+    resource = models.ForeignKey(
+        Resource, related_name="attributes", blank=True, null=True, on_delete=models.CASCADE
+    )  # TODO: why nullable ?
+
+    updated_at = models.DateTimeField(auto_now=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+
+class InputGroup(models.Model):
+    id_ = models.TextField(name="id", primary_key=True, default=cuid, editable=False)
+    merging_script = models.TextField(blank=True, default="")
+    attribute = models.ForeignKey(
+        Attribute, related_name="input_groups", blank=True, null=True, on_delete=models.CASCADE
+    )  # TODO: why nullable ?
+
+    updated_at = models.DateTimeField(auto_now=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+
+class Input(models.Model):
+    id_ = models.TextField(name="id", primary_key=True, default=cuid, editable=False)
+    input_group = models.ForeignKey(
+        InputGroup, related_name="inputs", null=True, on_delete=models.CASCADE
+    )  # TODO: why nullable ?
+    script = models.TextField(blank=True, default="")
+    concept_map_id = models.TextField(blank=True, default="")
+    static_value = models.TextField(blank=True, default="")
+
+    updated_at = models.DateTimeField(auto_now=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+
+class Column(models.Model):
+    id_ = models.TextField(name="id", primary_key=True, default=cuid, editable=False)
+    table = models.TextField()  # Expected to sometimes be null in prod
+    column = models.TextField()  # Expected to sometimes be null in prod
+    join = models.ForeignKey(
+        "Join", related_name="columns", blank=True, null=True, on_delete=models.CASCADE
+    )  # TODO: Why nullable ?
```

Comment:
a column may be part of a join (in this case it references the Join table) but it's optionnal, it should remain nullable